lib/resource/changes/create_new_version.ex

```diff
@@ -128,4 +130,38 @@ defmodule AshPaperTrail.Resource.Changes.CreateNewVersion do
     {:ok, dumped_value} = Ash.Type.dump_to_embedded(attribute.type, value, [])
     Map.put(changes, attribute.name, dumped_value)
   end
+
+  defp maybe_add_original_inputs(params, changeset) do
+    if AshPaperTrail.Resource.Info.store_inputs?(changeset.resource) do
+      inputs =
+        changeset.attributes
+        |> Map.merge(changeset.arguments)
+        |> normalise_inputs()
```

Comment:
We shouldn't do this by looking for structs, we should get the corresponding argument or attribute and use `dump_to_embedded` on it regardless of if the value is a struct or not.

Reply:
That will take care of the recursive aspect as well.

Reply:
Just this one open issue. Otherwise, looks good.